js_modules/dagit/packages/core/src/nav/WorkspaceStatus.tsx

```diff
@@ -0,0 +1,233 @@
+            {addedEntries.length === 1 ? (
+              <span>
+                Code location <strong>{addedEntries[0]}</strong> added
+              </span>
+            ) : (
+              <span>{addedEntries.length} code locations added</span>
+            )}
+            <ButtonLink onClick={() => history.push('/workspace')} color={Colors.White}>
+              View
+            </ButtonLink>
+          </Box>
+        ),
+        icon: 'add_circle',
+      });
+
+      reloadWorkspaceLoudly();
+      return;
+    }
+
+    const anyPreviouslyLoading = previousEntries.some(
+      ({loadStatus}) => loadStatus === RepositoryLocationLoadStatus.LOADING,
+    );
+
+    // One or more code locations are updating, so let the user know. We will not refetch the workspace
+    // until all code locations are done updating.
+    if (!anyPreviouslyLoading && anyCurrentlyLoading) {
+      setShowSpinner(true);
+
+      SharedToaster.show({
+        intent: 'primary',
+        message: (
+          <Box flex={{direction: 'row', justifyContent: 'space-between', grow: 1}}>
+            {currentlyLoading.length === 1 ? (
+              <span>
+                Updating <strong>{currentlyLoading[0].id}</strong>
+              </span>
+            ) : (
+              <span>Updating {currentlyLoading.length} code locations</span>
+            )}
+            <ButtonLink onClick={() => history.push('/workspace')} color={Colors.White}>
+              View
+            </ButtonLink>
+          </Box>
+        ),
+        icon: 'refresh',
+      });
+
+      return;
+    }
+
+    // A location was previously loading, and no longer is. Our workspace is ready. Refetch it.
+    if (anyPreviouslyLoading && !anyCurrentlyLoading) {
+      reloadWorkspaceLoudly();
+    }
+
+    // It's unlikely that we've made it to this point, since being inside this effect should
+    // indicate that `data` and `previousData` have differences that would have been handled by
+    // the conditionals above.
+  }, [data, previousData, reloadWorkspaceQuietly, reloadWorkspaceLoudly, history]);
+
+  if (showSpinner) {
+    return (
+      <Tooltip content="Reloading workspace…" placement="bottom">
+        <Spinner purpose="body-text" fillColor={Colors.Gray300} />
+      </Tooltip>
+    );
+  }
+
+  const repoErrors = locationEntries.filter(
+    (locationEntry) => locationEntry.locationOrLoadError?.__typename === 'PythonError',
+  );
+
+  if (repoErrors.length) {
+    return (
+      <WarningTooltip
+        content={
+          <div>{`${repoErrors.length} ${
+            repoErrors.length === 1
+              ? 'repository location failed to load'
+              : 'repository locations failed to load'
+          }`}</div>
+        }
+        position="bottom"
+        modifiers={{offset: {enabled: true, options: {offset: [0, 28]}}}}
+      >
+        <Icon name="warning" color={Colors.Yellow500} />
+      </WarningTooltip>
+    );
+  }
+
+  return <div style={{width: '16px'}} />;
```

Comment:
Is this `div` needed? Can you just return `null` or is this for spacing?

Reply:
It's for spacing, yeah.